### PR TITLE
Remove parenthesized expression

### PR DESCRIPTION
**Pull request by ghislainfourny**
ParenthesizedExpressions are also not expressions, they are a syntactic artifact. This cleans it up.

### CODE REVIEW

File: src/main/java/org/rumbledb/expressions/module/MainModule.java [hunk 1]
```diff
@@ -33,20 +33,20 @@
 public class MainModule extends Expression {
 
     private final Prolog prolog;
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/expressions/module/MainModule.java]: For completeness, should we also refactor Prolog type with Expression type here?

File: src/main/java/sparksoniq/jsoniq/compiler/JsoniqExpressionTreeVisitor.java [hunk 2]
```diff
@@ -109,7 +107,6 @@
     private Expression currentExpression;
     private Expression otherCurrentExpression;
 
-    private PrimaryExpression currentPrimaryExpression;
```
Comment (CanBerker) [inline src/main/java/sparksoniq/jsoniq/compiler/JsoniqExpressionTreeVisitor.java]: Nice !

Comment (ghislainfourny) [reply src/main/java/sparksoniq/jsoniq/compiler/JsoniqExpressionTreeVisitor.java]: I anticipated that you would like this one!

File: src/main/java/org/rumbledb/expressions/module/MainModule.java (hunk 1 above)
Comment (ghislainfourny) [reply src/main/java/org/rumbledb/expressions/module/MainModule.java]: Actually a prolog is not an expression (but this part is not refactored yet).

**Pull request merged**